plugins/inputs/influxdb_listener/influxdb_listener.go

```diff
@@ -0,0 +1,404 @@
+package influxdb_listener
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	tlsint "github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/selfstat"
+)
+
+const (
+	// defaultMaxBodySize is the default maximum request body size, in bytes.
+	// if the request body is over this size, we will return an HTTP 413 error.
+	defaultMaxBodySize = 32 * 1024 * 1024
+)
+
+type InfluxDBListener struct {
+	ServiceAddress string `toml:"service_address"`
+	port           int
+	tlsint.ServerConfig
+
+	ReadTimeout   internal.Duration `toml:"read_timeout"`
+	WriteTimeout  internal.Duration `toml:"write_timeout"`
+	MaxBodySize   internal.Size     `toml:"max_body_size"`
+	BasicUsername string            `toml:"basic_username"`
+	BasicPassword string            `toml:"basic_password"`
+	DatabaseTag   string            `toml:"database_tag"`
+
+	timeFunc influx.TimeFunc
+
+	listener net.Listener
+	server   http.Server
+
+	acc telegraf.Accumulator
+
+	bytesRecv       selfstat.Stat
+	requestsServed  selfstat.Stat
+	writesServed    selfstat.Stat
+	queriesServed   selfstat.Stat
+	pingsServed     selfstat.Stat
+	requestsRecv    selfstat.Stat
+	notFoundsServed selfstat.Stat
+	buffersCreated  selfstat.Stat
+	authFailures    selfstat.Stat
+
+	log telegraf.Logger
```

Comment:
It worked for me in the unit tests because I changed the tests to set lower case log.  I see now that it doesn't work like that.  :)  I've changed it like you suggested.